lib/prompts.js

```diff
@@ -1,197 +1,146 @@
 'use strict';
-
+const $ = exports;
 const el = require('./elements');
-const noop = () => {};
+const noop = v => v;
+
+function toPrompt(type, args, opts={}) {
+  return new Promise((res, rej) => {
+    const p = new el[type](args);
+    const onAbort = opts.onAbort || noop;
+    const onSubmit = opts.onSubmit || noop;
+    p.on('state', args.onState || noop);
+    p.on('submit', x => res(onSubmit(x)));
```

Comment:
Now all prompts also have individual `onAbort` and `onSubmit`. That's cool

Reply:
Individual prompt functions, yeah. But not the user-defined prompt objects.